lib/vsc/install/shared_setup.py

```diff
@@ -966,21 +1019,30 @@ def sanitize(name):
     """
     Transforms name into a sensible string for use in setup.cfg.
 
-    python- is prefixed in case of
-        enviroment variable VSC_INSTALL_PYTHON is set to 1 and either
+    enviroment variable VSC_RPM_PYTHON is set to 1 and either
+        name starts with key from PYTHON_BDIST_RPM_PREFIX_MAP
+            new name starts with value
+        python- is prefixed in case of
             name is not in hardcoded list NO_PREFIX_PYTHON_BDIST_RPM
             name starts with 'vsc'
+            and name does not start with python-
     """
     if isinstance(name, basestring):
-        p_p = (os.environ.get('VSC_RPM_PYTHON', False) and
-               ((name not in NO_PREFIX_PYTHON_BDIST_RPM)
-                or name.startswith('vsc')))
-        if p_p:
-            name = 'python-%s' % name
 
+        if os.environ.get('VSC_RPM_PYTHON', 'NOT_ONE') == '1':
+            # hardcoded prefix map
+            for pydep, rpmname in PYTHON_BDIST_RPM_PREFIX_MAP.items():
+                if name.startswith(pydep):
+                    return rpmname+name[len(pydep):]
+
+            # more sensible map
+            p_p = (not ([x for x in NO_PREFIX_PYTHON_BDIST_RPM if name.startswith(x)] or name.startswith('python-'))
+                   or name.startswith('vsc'))
+            if p_p:
+                name = 'python-%s' % name
         return name
-
-    return ",".join([sanitize(r) for r in name])
+    else:
```

Comment:
it shouldn't be needed per se, but I like how it makes the recursion condition more explicit.
Less confusion in the future when reading is ok to stay for me.